src/render_keyhook_hittable.c

```diff
@@ -0,0 +1,83 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   render_keyhook_hittable.c                          :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/12/16 11:50:32 by gwolf             #+#    #+#             */
+/*   Updated: 2023/12/23 22:40:51 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "render.h"
+
+void	ft_change_active_hittable(int key, t_entities *scene)
+{
+	if (scene->active == -1)
+		return ;
+	if (key == XK_n)
+	{
+		scene->active--;
+		if (scene->active < 0)
+			scene->active = scene->total - 1;
+	}
+	if (key == XK_m)
+	{
+		scene->active++;
+		if (scene->active == scene->total)
+			scene->active = 0;
+	}
+	ft_print_hittable(scene->obj[scene->active]);
+}
+
+void	ft_manip_hittable(int key, t_hittable *hittable)
+{
+	if (hittable->type == SPHERE)
+		ft_manip_sphere(key, &hittable->params.sp);
+	else if (hittable->type == PLANE)
+		ft_manip_plane(key, &hittable->params.pl);
+	else if (hittable->type == CYLINDER)
+		ft_manip_cylinder(key, &hittable->params.cy);
+	ft_print_hittable(*hittable);
+}
+
+
+void	ft_manip_sphere(int key, t_sphere *sp)
+{
+	if (key >= XK_0 && key <= XK_9)
+		ft_keyhook_change_col(key, &sp->colour);
+	else if (key == XK_r)
+		sp->r -= MV_UNIT;
```

Comment:
needs to be changed to sp->d, it was a bug in the previous code. We are given the diameter

Reply:
I think it makes more sense to leave it and change it when you merge your fix: I think it's more than just swapping the letter from r to d?

Reply:
In the calculation of the hit of the sphere the code needed to be adjusted, but that's the only instance next to of course the parsing. I already did that. As you want, we can wait.